catalog/transformer/catalog-transformer-preview/src/main/java/org/codice/ddf/transformer/preview/PreviewMetacardTransformer.java

```diff
@@ -72,24 +75,18 @@ public BinaryContent transform(Metacard metacard, Map<String, Serializable> argu
       }
     }
 
-    if (StringUtils.isNotEmpty(preview)) {
-      preview = String.format("<head><meta charset=\"utf-8\"/>%s</head>", preview);
-    }
+    preview = String.format("<head><meta charset=\"utf-8\"/>%s</head>", preview);
 
     return new BinaryContentImpl(IOUtils.toInputStream(preview));
   }
 
-  private String evaluateXPath(String metadata, String textElement) {
-    String xPathString = "//*[name()='%s']";
-    String xpath = String.format(xPathString, textElement);
-
+  private String evaluateXPath(String metadata, XPathExpression xPathExpression) {
     StringReader metadataReader = new StringReader(metadata);
     InputSource inputXml = new InputSource(metadataReader);
 
     String text = null;
     try {
-      Node result =
-          (Node) xPathFactory.newXPath().compile(xpath).evaluate(inputXml, XPathConstants.NODE);
+      Node result = (Node) xPathExpression.evaluate(inputXml, XPathConstants.NODE);
       if (result != null) {
         text = result.getTextContent().trim();
```

Comment:
❗️ This could still theoretically throw an NPE. I think we need to check `result.getTextContent()` for `null`.